addon/components/infinity-loader.js

```diff
@@ -77,22 +95,29 @@ const InfinityLoaderComponent = Ember.Component.extend({
   /**
    * is _scrollable's bottom position greater than the distance from the
     infinity-loader component to _scrollable's top
-   * in other words if at top of page and _scrollable's height is 100px and 
+   * in other words if at top of page and _scrollable's height is 100px and
     infinity loader is 200px from the top, has the scrollable traversed 101px to trigger loading more records
    * @method _shouldLoadMore
    * @return {Boolean}
    */
   _shouldLoadMore() {
-    if (this.get('developmentMode') || typeof FastBoot !== 'undefined' || this.isDestroying || this.isDestroyed) {
+    const consideredReady = !this.get('_isInfinityPromiseArray') || this.get('infinityModel.isFulfilled');
```

Comment:
The first part of this `||` statement - if it is not the PromiseProxy, then it isn't considered ready to load more?

Reply:
I guess I would have imagined only the `this.get('infinityModel.isFulfilled')` is needed.  But I may be missing something!

Reply:
in the case this is a "raw" infinity model (ie - not wrapped in InfinityPromiseArray) we consider it ready to go (that means the first set of content is resolved). this is the `!this.get('_isInfinityPromiseArray')` case

(also - In that case, `isFulfilled` is undefined - it's only available on PromiseProxies)

however:

in the case it's a PromiseArray, we need to be sure that the first load has actually resolved (so we wait until isFulfilled is true)